runtime/compiler/runtime/IProfiler.cpp

```diff
@@ -3925,16 +3905,18 @@ bool
 TR_IProfiler::postIprofilingBufferToWorkingQueue(J9VMThread * vmThread, const U_8* dataStart, UDATA size)
    {
    PORT_ACCESS_FROM_PORT(_portLib);
+
    //--- first acquire the monitor and try to get a free buffer
    if (!_iprofilerMonitor || _iprofilerMonitor->try_enter())
       return false; // Monitor is contended; better let the app thread do the processing
 
    // If the profiling thread has already been destroyed, delegate the processing to the java thread
-   if (_iprofilerThreadExitFlag)
+   if (getIProfilerThreadLifetimeState() == TR_IProfiler::IPROF_THR_DESTROYED)
```

Comment:
Technically yes, but I wrote the code this way so that the behaviour was the same as before. Namely, `_iprofilerThreadExitFlag` gets set by the IProfiler thread itself right before it calls `j9thread_exit`. In the previous code it seems possible for an app thread to post a new buffer between the point the shutdown thread calls `_iprofilerMonitor->wait()` and the IProfiler thread sets the `_iprofilerThreadExitFlag` flag.

I suppose we could have the app thread just return early if the state is STOPPING since it's not like the buffer will get processed anyway.